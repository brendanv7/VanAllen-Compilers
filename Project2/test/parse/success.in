{}$

{{{{{{}}}}}}$

{
  print("")
}$

{
  print(a)
}$

{
  print(2)
}$

{
  print("a")
}$

{
  print(false)
}$

{
  print("int a")
}$

{
  a = 1
}$

{
  a = 1 + 2 + 3 + 4   +      5
}$

{
  int a
  a = a
  string b
  a = b
}$

{ int a a=a string b a=b }$

{
  /* Comments? no problem */
  string b
}$

{
  string s
  s = "this string is /* in */ visible"
}$

{
  boolean d
  d = ("string" != "string")
}$

{
  while true {
    print ((true != 1 + 2))
  }
}
