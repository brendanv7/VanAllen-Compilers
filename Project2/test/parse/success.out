Compiling program 1...

LEXER -- Lexing program 1...
LEXER -- L_BRACE [ { ] found at (2:1)
LEXER -- R_BRACE [ } ] found at (2:2)
LEXER -- EOP [ $ ] found at (2:3)
LEXER -- Lexing complete for program 1. No errors found.

PARSER -- Parsing programNum 1...
PARSER -- parse()
PARSER -- parseprogramNum()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- Parse completed successfully.

CST for program 1:
<programNum>
-<Block>
--[{]
--<StatementList>
--[}]
-[$]

Compilation complete (for now) for program 1.

-------------------------------------------

Compiling program 2...

LEXER -- Lexing program 2...
LEXER -- L_BRACE [ { ] found at (5:1)
LEXER -- L_BRACE [ { ] found at (5:2)
LEXER -- L_BRACE [ { ] found at (5:3)
LEXER -- L_BRACE [ { ] found at (5:4)
LEXER -- L_BRACE [ { ] found at (5:5)
LEXER -- L_BRACE [ { ] found at (5:6)
LEXER -- R_BRACE [ } ] found at (5:7)
LEXER -- R_BRACE [ } ] found at (5:8)
LEXER -- R_BRACE [ } ] found at (5:9)
LEXER -- R_BRACE [ } ] found at (5:10)
LEXER -- R_BRACE [ } ] found at (5:11)
LEXER -- R_BRACE [ } ] found at (5:12)
LEXER -- EOP [ $ ] found at (5:13)
LEXER -- Lexing complete for program 2. No errors found.

PARSER -- Parsing programNum 2...
PARSER -- parse()
PARSER -- parseprogramNum()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatementList()
PARSER -- parseStatementList()
PARSER -- parseStatementList()
PARSER -- parseStatementList()
PARSER -- parseStatementList()
PARSER -- Parse completed successfully.

CST for program 2:
<programNum>
-<Block>
--[{]
--<StatementList>
---<Statement>
----<Block>
-----[{]
-----<StatementList>
------<Statement>
-------<Block>
--------[{]
--------<StatementList>
---------<Statement>
----------<Block>
-----------[{]
-----------<StatementList>
------------<Statement>
-------------<Block>
--------------[{]
--------------<StatementList>
---------------<Statement>
----------------<Block>
-----------------[{]
-----------------<StatementList>
-----------------[}]
---------------<StatementList>
--------------[}]
------------<StatementList>
-----------[}]
---------<StatementList>
--------[}]
------<StatementList>
-----[}]
---<StatementList>
--[}]
-[$]

Compilation complete (for now) for program 2.

-------------------------------------------

Compiling program 3...

LEXER -- Lexing program 3...
LEXER -- L_BRACE [ { ] found at (8:1)
LEXER -- PRINT [ print ] found at (10:3)
LEXER -- L_PAREN [ ( ] found at (10:8)
LEXER -- QUOTE [ " ] found at (10:9)
LEXER -- QUOTE [ " ] found at (10:10)
LEXER -- R_PAREN [ ) ] found at (10:11)
LEXER -- R_BRACE [ } ] found at (12:1)
LEXER -- EOP [ $ ] found at (12:2)
LEXER -- Lexing complete for program 3. No errors found.

PARSER -- Parsing programNum 3...
PARSER -- parse()
PARSER -- parseprogramNum()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parsePrintStatement()
PARSER -- parseExpr()
PARSER -- parseStringExpr()
PARSER -- parseCharList()
PARSER -- parseStatementList()
PARSER -- Parse completed successfully.

CST for program 3:
<programNum>
-<Block>
--[{]
--<StatementList>
---<Statement>
----<PrintStatement>
-----[print]
-----[(]
-----<Expr>
------<StringExpr>
-------["]
-------<CharList>
-------["]
-----[)]
---<StatementList>
--[}]
-[$]

Compilation complete (for now) for program 3.

-------------------------------------------

Compiling program 4...

LEXER -- Lexing program 4...
LEXER -- L_BRACE [ { ] found at (15:1)
LEXER -- PRINT [ print ] found at (17:3)
LEXER -- L_PAREN [ ( ] found at (17:8)
LEXER -- CHAR [ a ] found at (17:9)
LEXER -- R_PAREN [ ) ] found at (17:10)
LEXER -- R_BRACE [ } ] found at (19:1)
LEXER -- EOP [ $ ] found at (19:2)
LEXER -- Lexing complete for program 4. No errors found.

PARSER -- Parsing programNum 4...
PARSER -- parse()
PARSER -- parseprogramNum()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parsePrintStatement()
PARSER -- parseExpr()
PARSER -- parseId()
PARSER -- parseStatementList()
PARSER -- Parse completed successfully.

CST for program 4:
<programNum>
-<Block>
--[{]
--<StatementList>
---<Statement>
----<PrintStatement>
-----[print]
-----[(]
-----<Expr>
------<Id>
-------[a]
-----[)]
---<StatementList>
--[}]
-[$]

Compilation complete (for now) for program 4.

-------------------------------------------

Compiling program 5...

LEXER -- Lexing program 5...
LEXER -- L_BRACE [ { ] found at (22:1)
LEXER -- PRINT [ print ] found at (24:3)
LEXER -- L_PAREN [ ( ] found at (24:8)
LEXER -- DIGIT [ 2 ] found at (24:9)
LEXER -- R_PAREN [ ) ] found at (24:10)
LEXER -- R_BRACE [ } ] found at (26:1)
LEXER -- EOP [ $ ] found at (26:2)
LEXER -- Lexing complete for program 5. No errors found.

PARSER -- Parsing programNum 5...
PARSER -- parse()
PARSER -- parseprogramNum()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parsePrintStatement()
PARSER -- parseExpr()
PARSER -- parseIntExpr()
PARSER -- parseStatementList()
PARSER -- Parse completed successfully.

CST for program 5:
<programNum>
-<Block>
--[{]
--<StatementList>
---<Statement>
----<PrintStatement>
-----[print]
-----[(]
-----<Expr>
------<IntExpr>
-------[2]
-----[)]
---<StatementList>
--[}]
-[$]

Compilation complete (for now) for program 5.

-------------------------------------------

Compiling program 6...

LEXER -- Lexing program 6...
LEXER -- L_BRACE [ { ] found at (29:1)
LEXER -- PRINT [ print ] found at (31:3)
LEXER -- L_PAREN [ ( ] found at (31:8)
LEXER -- QUOTE [ " ] found at (31:9)
LEXER -- CHAR [ a ] found at (31:10)
LEXER -- QUOTE [ " ] found at (31:11)
LEXER -- R_PAREN [ ) ] found at (31:12)
LEXER -- R_BRACE [ } ] found at (33:1)
LEXER -- EOP [ $ ] found at (33:2)
LEXER -- Lexing complete for program 6. No errors found.

PARSER -- Parsing programNum 6...
PARSER -- parse()
PARSER -- parseprogramNum()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parsePrintStatement()
PARSER -- parseExpr()
PARSER -- parseStringExpr()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseStatementList()
PARSER -- Parse completed successfully.

CST for program 6:
<programNum>
-<Block>
--[{]
--<StatementList>
---<Statement>
----<PrintStatement>
-----[print]
-----[(]
-----<Expr>
------<StringExpr>
-------["]
-------<CharList>
--------[a]
--------<CharList>
-------["]
-----[)]
---<StatementList>
--[}]
-[$]

Compilation complete (for now) for program 6.

-------------------------------------------

Compiling program 7...

LEXER -- Lexing program 7...
LEXER -- L_BRACE [ { ] found at (36:1)
LEXER -- PRINT [ print ] found at (38:3)
LEXER -- L_PAREN [ ( ] found at (38:8)
LEXER -- BOOL_VAL [ false ] found at (38:9)
LEXER -- R_PAREN [ ) ] found at (38:14)
LEXER -- R_BRACE [ } ] found at (40:1)
LEXER -- EOP [ $ ] found at (40:2)
LEXER -- Lexing complete for program 7. No errors found.

PARSER -- Parsing programNum 7...
PARSER -- parse()
PARSER -- parseprogramNum()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parsePrintStatement()
PARSER -- parseExpr()
PARSER -- parseBooleanExpr()
PARSER -- parseStatementList()
PARSER -- Parse completed successfully.

CST for program 7:
<programNum>
-<Block>
--[{]
--<StatementList>
---<Statement>
----<PrintStatement>
-----[print]
-----[(]
-----<Expr>
------<BooleanExpr>
-------[false]
-----[)]
---<StatementList>
--[}]
-[$]

Compilation complete (for now) for program 7.

-------------------------------------------

Compiling program 8...

LEXER -- Lexing program 8...
LEXER -- L_BRACE [ { ] found at (43:1)
LEXER -- PRINT [ print ] found at (45:3)
LEXER -- L_PAREN [ ( ] found at (45:8)
LEXER -- QUOTE [ " ] found at (45:9)
LEXER -- CHAR [ i ] found at (45:10)
LEXER -- CHAR [ n ] found at (45:11)
LEXER -- CHAR [ t ] found at (45:12)
LEXER -- SPACE [   ] found at (45:13)
LEXER -- CHAR [ a ] found at (45:14)
LEXER -- QUOTE [ " ] found at (45:15)
LEXER -- R_PAREN [ ) ] found at (45:16)
LEXER -- R_BRACE [ } ] found at (47:1)
LEXER -- EOP [ $ ] found at (47:2)
LEXER -- Lexing complete for program 8. No errors found.

PARSER -- Parsing programNum 8...
PARSER -- parse()
PARSER -- parseprogramNum()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parsePrintStatement()
PARSER -- parseExpr()
PARSER -- parseStringExpr()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseStatementList()
PARSER -- Parse completed successfully.

CST for program 8:
<programNum>
-<Block>
--[{]
--<StatementList>
---<Statement>
----<PrintStatement>
-----[print]
-----[(]
-----<Expr>
------<StringExpr>
-------["]
-------<CharList>
--------[i]
--------<CharList>
---------[n]
---------<CharList>
----------[t]
----------<CharList>
-----------[ ]
-----------<CharList>
------------[a]
------------<CharList>
-------["]
-----[)]
---<StatementList>
--[}]
-[$]

Compilation complete (for now) for program 8.

-------------------------------------------

Compiling program 9...

LEXER -- Lexing program 9...
LEXER -- L_BRACE [ { ] found at (50:1)
LEXER -- CHAR [ a ] found at (52:3)
LEXER -- ASSIGN_OP [ = ] found at (52:5)
LEXER -- DIGIT [ 1 ] found at (52:7)
LEXER -- R_BRACE [ } ] found at (54:1)
LEXER -- EOP [ $ ] found at (54:2)
LEXER -- Lexing complete for program 9. No errors found.

PARSER -- Parsing programNum 9...
PARSER -- parse()
PARSER -- parseprogramNum()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseAssignmentStatement()
PARSER -- parseId()
PARSER -- parseExpr()
PARSER -- parseIntExpr()
PARSER -- parseStatementList()
PARSER -- Parse completed successfully.

CST for program 9:
<programNum>
-<Block>
--[{]
--<StatementList>
---<Statement>
----<AssignmentStatement>
-----<Id>
------[a]
-----[=]
-----<Expr>
------<IntExpr>
-------[1]
---<StatementList>
--[}]
-[$]

Compilation complete (for now) for program 9.

-------------------------------------------

Compiling program 10...

LEXER -- Lexing program 10...
LEXER -- L_BRACE [ { ] found at (57:1)
LEXER -- CHAR [ a ] found at (59:3)
LEXER -- ASSIGN_OP [ = ] found at (59:5)
LEXER -- DIGIT [ 1 ] found at (59:7)
LEXER -- INT_OP [ + ] found at (59:9)
LEXER -- DIGIT [ 2 ] found at (59:11)
LEXER -- INT_OP [ + ] found at (59:13)
LEXER -- DIGIT [ 3 ] found at (59:15)
LEXER -- INT_OP [ + ] found at (59:17)
LEXER -- DIGIT [ 4 ] found at (59:19)
LEXER -- INT_OP [ + ] found at (59:23)
LEXER -- DIGIT [ 5 ] found at (59:30)
LEXER -- R_BRACE [ } ] found at (61:1)
LEXER -- EOP [ $ ] found at (61:2)
LEXER -- Lexing complete for program 10. No errors found.

PARSER -- Parsing programNum 10...
PARSER -- parse()
PARSER -- parseprogramNum()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseAssignmentStatement()
PARSER -- parseId()
PARSER -- parseExpr()
PARSER -- parseIntExpr()
PARSER -- parseExpr()
PARSER -- parseIntExpr()
PARSER -- parseExpr()
PARSER -- parseIntExpr()
PARSER -- parseExpr()
PARSER -- parseIntExpr()
PARSER -- parseExpr()
PARSER -- parseIntExpr()
PARSER -- parseStatementList()
PARSER -- Parse completed successfully.

CST for program 10:
<programNum>
-<Block>
--[{]
--<StatementList>
---<Statement>
----<AssignmentStatement>
-----<Id>
------[a]
-----[=]
-----<Expr>
------<IntExpr>
-------[1]
-------[+]
-------<Expr>
--------<IntExpr>
---------[2]
---------[+]
---------<Expr>
----------<IntExpr>
-----------[3]
-----------[+]
-----------<Expr>
------------<IntExpr>
-------------[4]
-------------[+]
-------------<Expr>
--------------<IntExpr>
---------------[5]
---<StatementList>
--[}]
-[$]

Compilation complete (for now) for program 10.

-------------------------------------------

Compiling program 11...

LEXER -- Lexing program 11...
LEXER -- L_BRACE [ { ] found at (64:1)
LEXER -- INT [ int ] found at (66:3)
LEXER -- CHAR [ a ] found at (66:7)
LEXER -- CHAR [ a ] found at (68:3)
LEXER -- ASSIGN_OP [ = ] found at (68:5)
LEXER -- CHAR [ a ] found at (68:7)
LEXER -- STRING [ string ] found at (70:3)
LEXER -- CHAR [ b ] found at (70:10)
LEXER -- CHAR [ a ] found at (72:3)
LEXER -- ASSIGN_OP [ = ] found at (72:5)
LEXER -- CHAR [ b ] found at (72:7)
LEXER -- R_BRACE [ } ] found at (74:1)
LEXER -- EOP [ $ ] found at (74:2)
LEXER -- Lexing complete for program 11. No errors found.

PARSER -- Parsing programNum 11...
PARSER -- parse()
PARSER -- parseprogramNum()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseVarDecl()
PARSER -- parseId()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseAssignmentStatement()
PARSER -- parseId()
PARSER -- parseExpr()
PARSER -- parseId()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseVarDecl()
PARSER -- parseId()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseAssignmentStatement()
PARSER -- parseId()
PARSER -- parseExpr()
PARSER -- parseId()
PARSER -- parseStatementList()
PARSER -- Parse completed successfully.

CST for program 11:
<programNum>
-<Block>
--[{]
--<StatementList>
---<Statement>
----<VarDecl>
-----[int]
-----<Id>
------[a]
---<StatementList>
----<Statement>
-----<AssignmentStatement>
------<Id>
-------[a]
------[=]
------<Expr>
-------<Id>
--------[a]
----<StatementList>
-----<Statement>
------<VarDecl>
-------[string]
-------<Id>
--------[b]
-----<StatementList>
------<Statement>
-------<AssignmentStatement>
--------<Id>
---------[a]
--------[=]
--------<Expr>
---------<Id>
----------[b]
------<StatementList>
--[}]
-[$]

Compilation complete (for now) for program 11.

-------------------------------------------

Compiling program 12...

LEXER -- Lexing program 12...
LEXER -- L_BRACE [ { ] found at (77:1)
LEXER -- INT [ int ] found at (77:3)
LEXER -- CHAR [ a ] found at (77:7)
LEXER -- CHAR [ a ] found at (77:9)
LEXER -- ASSIGN_OP [ = ] found at (77:10)
LEXER -- CHAR [ a ] found at (77:11)
LEXER -- STRING [ string ] found at (77:13)
LEXER -- CHAR [ b ] found at (77:20)
LEXER -- CHAR [ a ] found at (77:22)
LEXER -- ASSIGN_OP [ = ] found at (77:23)
LEXER -- CHAR [ b ] found at (77:24)
LEXER -- R_BRACE [ } ] found at (77:26)
LEXER -- EOP [ $ ] found at (77:27)
LEXER -- Lexing complete for program 12. No errors found.

PARSER -- Parsing programNum 12...
PARSER -- parse()
PARSER -- parseprogramNum()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseVarDecl()
PARSER -- parseId()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseAssignmentStatement()
PARSER -- parseId()
PARSER -- parseExpr()
PARSER -- parseId()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseVarDecl()
PARSER -- parseId()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseAssignmentStatement()
PARSER -- parseId()
PARSER -- parseExpr()
PARSER -- parseId()
PARSER -- parseStatementList()
PARSER -- Parse completed successfully.

CST for program 12:
<programNum>
-<Block>
--[{]
--<StatementList>
---<Statement>
----<VarDecl>
-----[int]
-----<Id>
------[a]
---<StatementList>
----<Statement>
-----<AssignmentStatement>
------<Id>
-------[a]
------[=]
------<Expr>
-------<Id>
--------[a]
----<StatementList>
-----<Statement>
------<VarDecl>
-------[string]
-------<Id>
--------[b]
-----<StatementList>
------<Statement>
-------<AssignmentStatement>
--------<Id>
---------[a]
--------[=]
--------<Expr>
---------<Id>
----------[b]
------<StatementList>
--[}]
-[$]

Compilation complete (for now) for program 12.

-------------------------------------------

Compiling program 13...

LEXER -- Lexing program 13...
LEXER -- L_BRACE [ { ] found at (80:1)
LEXER -- STRING [ string ] found at (84:3)
LEXER -- CHAR [ b ] found at (84:10)
LEXER -- R_BRACE [ } ] found at (86:1)
LEXER -- EOP [ $ ] found at (86:2)
LEXER -- Lexing complete for program 13. No errors found.

PARSER -- Parsing programNum 13...
PARSER -- parse()
PARSER -- parseprogramNum()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseVarDecl()
PARSER -- parseId()
PARSER -- parseStatementList()
PARSER -- Parse completed successfully.

CST for program 13:
<programNum>
-<Block>
--[{]
--<StatementList>
---<Statement>
----<VarDecl>
-----[string]
-----<Id>
------[b]
---<StatementList>
--[}]
-[$]

Compilation complete (for now) for program 13.

-------------------------------------------

Compiling program 14...

LEXER -- Lexing program 14...
LEXER -- L_BRACE [ { ] found at (89:1)
LEXER -- STRING [ string ] found at (91:3)
LEXER -- CHAR [ s ] found at (91:10)
LEXER -- CHAR [ s ] found at (93:3)
LEXER -- ASSIGN_OP [ = ] found at (93:5)
LEXER -- QUOTE [ " ] found at (93:7)
LEXER -- CHAR [ t ] found at (93:8)
LEXER -- CHAR [ h ] found at (93:9)
LEXER -- CHAR [ i ] found at (93:10)
LEXER -- CHAR [ s ] found at (93:11)
LEXER -- SPACE [   ] found at (93:12)
LEXER -- CHAR [ s ] found at (93:13)
LEXER -- CHAR [ t ] found at (93:14)
LEXER -- CHAR [ r ] found at (93:15)
LEXER -- CHAR [ i ] found at (93:16)
LEXER -- CHAR [ n ] found at (93:17)
LEXER -- CHAR [ g ] found at (93:18)
LEXER -- SPACE [   ] found at (93:19)
LEXER -- CHAR [ i ] found at (93:20)
LEXER -- CHAR [ s ] found at (93:21)
LEXER -- SPACE [   ] found at (93:22)
LEXER -- SPACE [   ] found at (93:29)
LEXER -- CHAR [ v ] found at (93:32)
LEXER -- CHAR [ i ] found at (93:33)
LEXER -- CHAR [ s ] found at (93:34)
LEXER -- CHAR [ i ] found at (93:35)
LEXER -- CHAR [ b ] found at (93:36)
LEXER -- CHAR [ l ] found at (93:37)
LEXER -- CHAR [ e ] found at (93:38)
LEXER -- QUOTE [ " ] found at (93:39)
LEXER -- R_BRACE [ } ] found at (95:1)
LEXER -- EOP [ $ ] found at (95:2)
LEXER -- Lexing complete for program 14. No errors found.

PARSER -- Parsing programNum 14...
PARSER -- parse()
PARSER -- parseprogramNum()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseVarDecl()
PARSER -- parseId()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseAssignmentStatement()
PARSER -- parseId()
PARSER -- parseExpr()
PARSER -- parseStringExpr()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseStatementList()
PARSER -- Parse completed successfully.

CST for program 14:
<programNum>
-<Block>
--[{]
--<StatementList>
---<Statement>
----<VarDecl>
-----[string]
-----<Id>
------[s]
---<StatementList>
----<Statement>
-----<AssignmentStatement>
------<Id>
-------[s]
------[=]
------<Expr>
-------<StringExpr>
--------["]
--------<CharList>
---------[t]
---------<CharList>
----------[h]
----------<CharList>
-----------[i]
-----------<CharList>
------------[s]
------------<CharList>
-------------[ ]
-------------<CharList>
--------------[s]
--------------<CharList>
---------------[t]
---------------<CharList>
----------------[r]
----------------<CharList>
-----------------[i]
-----------------<CharList>
------------------[n]
------------------<CharList>
-------------------[g]
-------------------<CharList>
--------------------[ ]
--------------------<CharList>
---------------------[i]
---------------------<CharList>
----------------------[s]
----------------------<CharList>
-----------------------[ ]
-----------------------<CharList>
------------------------[ ]
------------------------<CharList>
-------------------------[v]
-------------------------<CharList>
--------------------------[i]
--------------------------<CharList>
---------------------------[s]
---------------------------<CharList>
----------------------------[i]
----------------------------<CharList>
-----------------------------[b]
-----------------------------<CharList>
------------------------------[l]
------------------------------<CharList>
-------------------------------[e]
-------------------------------<CharList>
--------["]
----<StatementList>
--[}]
-[$]

Compilation complete (for now) for program 14.

-------------------------------------------

Compiling program 15...

LEXER -- Lexing program 15...
LEXER -- L_BRACE [ { ] found at (98:1)
LEXER -- BOOL [ boolean ] found at (100:3)
LEXER -- CHAR [ d ] found at (100:11)
LEXER -- CHAR [ d ] found at (102:3)
LEXER -- ASSIGN_OP [ = ] found at (102:5)
LEXER -- L_PAREN [ ( ] found at (102:7)
LEXER -- QUOTE [ " ] found at (102:8)
LEXER -- CHAR [ s ] found at (102:9)
LEXER -- CHAR [ t ] found at (102:10)
LEXER -- CHAR [ r ] found at (102:11)
LEXER -- CHAR [ i ] found at (102:12)
LEXER -- CHAR [ n ] found at (102:13)
LEXER -- CHAR [ g ] found at (102:14)
LEXER -- QUOTE [ " ] found at (102:15)
LEXER -- BOOL_OP [ != ] found at (102:17)
LEXER -- QUOTE [ " ] found at (102:20)
LEXER -- CHAR [ s ] found at (102:21)
LEXER -- CHAR [ t ] found at (102:22)
LEXER -- CHAR [ r ] found at (102:23)
LEXER -- CHAR [ i ] found at (102:24)
LEXER -- CHAR [ n ] found at (102:25)
LEXER -- CHAR [ g ] found at (102:26)
LEXER -- QUOTE [ " ] found at (102:27)
LEXER -- R_PAREN [ ) ] found at (102:28)
LEXER -- R_BRACE [ } ] found at (104:1)
LEXER -- EOP [ $ ] found at (104:2)
LEXER -- Lexing complete for program 15. No errors found.

PARSER -- Parsing programNum 15...
PARSER -- parse()
PARSER -- parseprogramNum()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseVarDecl()
PARSER -- parseId()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseAssignmentStatement()
PARSER -- parseId()
PARSER -- parseExpr()
PARSER -- parseBooleanExpr()
PARSER -- parseExpr()
PARSER -- parseStringExpr()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseExpr()
PARSER -- parseStringExpr()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseCharList()
PARSER -- parseStatementList()
PARSER -- Parse completed successfully.

CST for program 15:
<programNum>
-<Block>
--[{]
--<StatementList>
---<Statement>
----<VarDecl>
-----[boolean]
-----<Id>
------[d]
---<StatementList>
----<Statement>
-----<AssignmentStatement>
------<Id>
-------[d]
------[=]
------<Expr>
-------<BooleanExpr>
--------[(]
--------<Expr>
---------<StringExpr>
----------["]
----------<CharList>
-----------[s]
-----------<CharList>
------------[t]
------------<CharList>
-------------[r]
-------------<CharList>
--------------[i]
--------------<CharList>
---------------[n]
---------------<CharList>
----------------[g]
----------------<CharList>
----------["]
--------[!=]
--------<Expr>
---------<StringExpr>
----------["]
----------<CharList>
-----------[s]
-----------<CharList>
------------[t]
------------<CharList>
-------------[r]
-------------<CharList>
--------------[i]
--------------<CharList>
---------------[n]
---------------<CharList>
----------------[g]
----------------<CharList>
----------["]
--------[)]
----<StatementList>
--[}]
-[$]

Compilation complete (for now) for program 15.

-------------------------------------------

Compiling program 16...

LEXER -- Lexing program 16...
LEXER -- L_BRACE [ { ] found at (107:1)
LEXER -- WHILE [ while ] found at (109:3)
LEXER -- BOOL_VAL [ true ] found at (109:9)
LEXER -- L_BRACE [ { ] found at (109:14)
LEXER -- PRINT [ print ] found at (111:5)
LEXER -- L_PAREN [ ( ] found at (111:11)
LEXER -- L_PAREN [ ( ] found at (111:12)
LEXER -- BOOL_VAL [ true ] found at (111:13)
LEXER -- BOOL_OP [ != ] found at (111:18)
LEXER -- DIGIT [ 1 ] found at (111:21)
LEXER -- INT_OP [ + ] found at (111:23)
LEXER -- DIGIT [ 2 ] found at (111:25)
LEXER -- R_PAREN [ ) ] found at (111:26)
LEXER -- R_PAREN [ ) ] found at (111:27)
LEXER -- R_BRACE [ } ] found at (113:3)
LEXER -- R_BRACE [ } ] found at (115:1)
LEXER -- WARNING: No EOP character found and end-of-file has been reached. at (116:0)
LEXER -- Lexing complete for program 16. No errors found.

PARSER -- Parsing programNum 16...
PARSER -- parse()
PARSER -- parseprogramNum()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parseWhileStatement()
PARSER -- parseBooleanExpr()
PARSER -- parseBlock()
PARSER -- parseStatementList()
PARSER -- parseStatement()
PARSER -- parsePrintStatement()
PARSER -- parseExpr()
PARSER -- parseBooleanExpr()
PARSER -- parseExpr()
PARSER -- parseBooleanExpr()
PARSER -- parseExpr()
PARSER -- parseIntExpr()
PARSER -- parseExpr()
PARSER -- parseIntExpr()
PARSER -- parseStatementList()
PARSER -- parseStatementList()
PARSER -- Parse completed successfully.

CST for program 16:
<programNum>
-<Block>
--[{]
--<StatementList>
---<Statement>
----<WhileStatement>
-----[while]
-----<BooleanExpr>
------[true]
-----<Block>
------[{]
------<StatementList>
-------<Statement>
--------<PrintStatement>
---------[print]
---------[(]
---------<Expr>
----------<BooleanExpr>
-----------[(]
-----------<Expr>
------------<BooleanExpr>
-------------[true]
-----------[!=]
-----------<Expr>
------------<IntExpr>
-------------[1]
-------------[+]
-------------<Expr>
--------------<IntExpr>
---------------[2]
-----------[)]
---------[)]
-------<StatementList>
------[}]
---<StatementList>
--[}]
-[$]

Compilation complete (for now) for program 16.

-------------------------------------------

